gc/base/TLHAllocationSupport.cpp

```diff
@@ -288,6 +295,13 @@ MM_TLHAllocationSupport::refresh(MM_EnvironmentBase *env, MM_AllocateDescription
 			if (getRefreshSize() < tlhMaximumSize) {
 				setRefreshSize(getRefreshSize() + extensions->tlhIncrementSize);
 			}
+
+			Assert_MM_true(_reservedBytes == 0);
+
+			if (sizeInBytesToReserve > 0) {
+				_reservedBytes = sizeInBytesToReserve;
+				setTop((void *) ((uintptr_t) getTop() - _reservedBytes));
+			}
```

Comment:
this might go into a separate helper

reserveTLHTop(ForGC) (symmetrical to restoreTLHTop)